### DISCUSSION TITLE
WiX5: Setup creates an extra folder when not using INSTALLFOLDER property at all.

### DISCUSSION THREAD

**Discussion by madmaxoft** [wixtoolset/issues#8781] [Questions] [answered]
My scripts don't have any reference to `INSTALLFOLDER` at all - I'm using a separate property name for the install location. Yet when I build the setup and run it, it contains an `INSTALLFOLDER` directory entry (as confirmed with the Orca editor) and it creates an empty folder `c:\Program Files\<manufacturer> <product>`. 

I've found some references to `INSTALLFOLDER` in the issues here on WiX Github, they speak of WiX auto-creating the folder reference when it is referenced elsewhere in the script files. But there's no such reference in my scripts. How can I turn this auto-creation off, then?

I've recently migrated my setup scripts from WiX3 to WiX5, in the old version the empty folder was not created. I'm using the commandline `wix build ...` invocation to build my setup (from CMake).

**Comment by wmanning** [comment]
Feature described here: https://wixtoolset.org/docs/fivefour/#default-installation-folder

**Reply by madmaxoft** [reply]
Yes, that part also says
> And don't worry -- if you already have such an `INSTALLFOLDER` or never reference a directory with that id, WiX respects your beliefs and won't try to force its own `INSTALLFOLDER` on you or your package.

Yet despite not having any `INSTALLFOLDER` in my WiX scripts, it is creating the default one.

**Reply by robmen** [reply, marked as answer]
`INSTALLFOLDER` _should_ only be added if something referenced it. Try opening your MSI in Orca and seeing if there are any references to `INSTALLFOLDER` that surprise you. Maybe one exists that you didn't think. If not, then it is possible there is a bug, and you'll need to debug deeper.

**Reply by madmaxoft** [reply]
Since there's no mention of `INSTALLFOLDER` anywhere in my scripts, I'd expect that it's not referenced.

Using Orca, I found out that the reference is silently created for `<Component>` elements without a `Directory` attribute (containing only registry entries, so no relation to filesystem at all). Once I added the attribute to all such Components, the auto-generated `INSTALLFOLDER` truly went away.

Thanks for pointing me in the right direction.

**Reply by robmen** [reply]
That's the new feature in v5. It allows you to write .wxs code like:

```xml
<Wix>
  <Package ...>
      <File Source="installthis.txt" />
  </Package>
</Wix>
```

And the defaults (like this new `INSTALLFOLDER` feature) give you everything you need to create a full-blown MSI.

**Comment by madmaxoft** [comment]
So far I've only come up with a workaround, but it's not perfect. I add this to my script:
```
<StandardDirectory Id="ProgramFiles6432Folder">
	<Directory Id="INSTALLFOLDER" Name="{my default install folder}" />
</StandardDirectory>
```
This works by replacing the bad empty folder with the default install folder, but I'm afraid it'll break if the user decides to install to a non-default folder.
